src/ape_ethereum/multicall/handlers.py

```diff
@@ -0,0 +1,262 @@
+            call: :class:`ContractMethodHandler` The method to call.
+            *args: The arguments to invoke the method with.
+            requireSuccess: bool Whether the call must be successful.
+            value: int The amount of ether to forward with the call.
+        """
+
+        # Update call type if relevant kwarg is used
+        if value is not None and self._handler_method_abi != AGGREGATE3VALUE_METHOD:
+            self._handler_method_abi = AGGREGATE3VALUE_METHOD
+
+            # Update all previous calls to use AGGREGATE3VALUE_METHOD
+            self.calls = [
+                (call[0], call[1], 0, call[2])  # type: ignore[misc]
+                if len(call) == 3
+                else (call[0], True, 0, call[1])
+                for call in self.calls
+            ]
+
+        elif requireSuccess is not None and self._handler_method_abi == AGGREGATE_METHOD:
+            self._handler_method_abi = AGGREGATE_METHOD
+
+            # Update all previous calls to use AGGREGATE3_METHOD
+            self.calls = [(call[0], True, call[1]) for call in self.calls]  # type: ignore[misc]
+
+        # Append call to the list
+        if self._handler_method_abi == AGGREGATE_METHOD:
+            self.calls.append((call.contract.address, call.encode_input(*args)))
+
+        elif self._handler_method_abi == AGGREGATE3_METHOD:
+            self.calls.append(
+                (
+                    call.contract.address,
+                    requireSuccess if requireSuccess is not None else True,
+                    call.encode_input(*args),
+                )
+            )
+
+        elif self._handler_method_abi == AGGREGATE3VALUE_METHOD:
+            self.calls.append(
+                (
+                    call.contract.address,
+                    requireSuccess if requireSuccess is not None else True,
+                    value or 0,
+                    call.encode_input(*args),
+                )
+            )
+
+
+class Call(BaseMulticall):
+    """
+    Create a sequence of calls to execute at once using `eth_call` via the Multicall3 contract.
+
+    Usage example::
+
+        from ape_ethereum import multicall
+
+        call = multicall.Call()
+        call.add(contract.myMethod, *call_args)
+        call.add(contract.myMethod, *call_args)
+        ...  # Add as many calls as desired
+        call.add(contract.myMethod, *call_args)
+        a, b, ..., z = call()  # Performs multicall
+    """
+
+    def __init__(self) -> None:
+        super().__init__(ContractCallHandler)
+
+        self.abis: List[MethodABI] = []
+        self._result: Union[None, Tuple[int, List[HexBytes]], List[Tuple[bool, HexBytes]]] = None
+
+    def add(self, call: ContractMethodHandler, *args, **kwargs):
+        if "value" in kwargs:
+            raise InvalidOption("value")
+
+        super().add(call, *args, **kwargs)
+        self.abis.append(_select_method_abi(call.abis, args))
+
+    @property
+    def returnData(self) -> List[HexBytes]:
```

Comment:
Or is this supposed to be camelCase?

```suggestion
    def return_data(self) -> List[HexBytes]:
```

Reply:
Yeah this one is camelCase to match the method's return value